util/internal/test/common/src/main/java/org/hibernate/search/util/impl/test/extension/parameterized/ParameterizedClassExtension.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.util.impl.test.extension.parameterized;
+
+import static java.util.Spliterators.spliteratorUnknownSize;
+import static java.util.stream.StreamSupport.stream;
+import static org.hibernate.search.util.impl.test.extension.parameterized.ParameterizedClassTestMethodDiscoverer.discover;
+import static org.junit.platform.commons.util.AnnotationUtils.isAnnotated;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+
+final class ParameterizedClassExtension
+		implements TestTemplateInvocationContextProvider, AfterTestExecutionCallback, ParameterResolver {
+	private List<Object[]> envArguments;
+	private int envIndex = 0;
+	private boolean envInitialized = false;
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+			throws ParameterResolutionException {
+		return true;
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+			throws ParameterResolutionException {
+		return envArguments.get( envIndex )[parameterContext.getIndex()];
+	}
+
+	@Override
+	public boolean supportsTestTemplate(ExtensionContext context) {
+		return isAnnotated( context.getTestMethod(), ParameterizedSetup.class );
+	}
+
+	@Override
+	public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(ExtensionContext context) {
```

Comment:
Should this throw an exception if the test class is not annotated with `@ParameterizedClass`?

Reply:
yeah, I'll have a check for it in `ParameterizedClassPostDiscoveryFilter`